### PR TITLE
[#5][UI] As a user, I can login

### LINKED ISSUE #5: [UI] As a user, I can login
## Why

Before the user can take the survey, they have to login with their email and password first.

## Acceptance Criteria

- When the user lands on the site, they should be able to see the login form.

## Resources

- [[Figma] Login form](https://www.figma.com/file/i1AI0IzCWuJApX5CK0DmiW/Challenge---Web-App?node-id=6%3A168)

### PR DESCRIPTION
**Pull request by rosle**
Closes https://github.com/rosle/react-nimble-survey/issues/5

## What happened 👀

- Implement a Form component
- Show the login form on login page

## Insight 📝

Use [React Hook Form](https://react-hook-form.com/get-started) to store form state.

## Proof Of Work 📹

https://user-images.githubusercontent.com/6965195/186154963-493ef30c-a518-4ecf-9b6e-1af51bdd7d1d.mov

### CODE REVIEW

File: src/components/Form/index.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  errors?: FormErrors;
+}
+
+const Form = ({ children, errors, ...formAttributes }: FormProps) => {
+  const { t } = useTranslation(['shared']);
+
+  const errorMessage = (formErrors: FormErrors): React.ReactNode => {
+    if (typeof formErrors === 'string') return formErrors;
+
+    const errorList = _.chain(formErrors)
```
Comment (hanam1ni) [inline src/components/Form/index.tsx]: Good to know about this function that can turn functions into pipeline 👍

File: src/components/Form/index.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    );
+
+    cy.findByTestId(formTestIds.form).should('be.visible');
+
+    cy.findByTestId(formTestIds.form).within(() => {
+      cy.findByTestId(inputTestIds.input).should('be.visible');
+    });
+  });
+
+  it('given there are no form errors, does NOT render form error', () => {
```
Comment (hanam1ni) [inline src/components/Form/index.cy.tsx]: Let's refactor the test description in this PR to put the context inside describe block

```
describe('given there are no form errors', () => {
  it(' ... ', () => {})
})
```

Comment (rosle) [reply src/components/Form/index.cy.tsx]: Updated 👍 - [3d02646](https://github.com/rosle/react-nimble-survey/pull/25/commits/3d0264612ae6a92be1a08fb786b316156eb2adf1)

File: src/screens/Login/index.test.tsx [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
   it('renders the AuthLayout with the correct title', () => {
     render(<LoginScreen />);
 
     const authLayoutHeaderTitle = screen.getByTestId(authLayoutTestIds.headerTitle);
 
-    expect(authLayoutHeaderTitle).toHaveTextContent('auth:sign_in');
+    expect(authLayoutHeaderTitle).toHaveTextContent('auth:heading.sign_in');
+  });
+
+  describe('given the invalid inputs', () => {
```
Comment (Lahphim) [inline src/screens/Login/index.test.tsx]:
```suggestion
  describe('given the INVALID inputs', () => {
```
Should we use uppercase for the happy word?

Comment (Lahphim) [inline src/screens/Login/index.test.tsx]: Should we add the valid case to just check that all error messages disappear?

```
describe('given the valid inputs', () => { ... }
```

Comment (rosle) [reply src/screens/Login/index.test.tsx]: Updated on [ecadfb6](https://github.com/rosle/react-nimble-survey/pull/25/commits/ecadfb62f51c5006868fa9084c49f2871979ee35) 👍

Comment (rosle) [reply src/screens/Login/index.test.tsx]: Updated on https://github.com/rosle/react-nimble-survey/pull/25/commits/503ef97d8cae65cd2211e06f619e69390630ac94 👍

**Review by hanam1ni — APPROVED** [review]

**Review by Lahphim — APPROVED** [review]

**Pull request merged**